File: sources/command/delete.py
from typing import NoReturn

import logging
import os

from globals import ExitCode, get_drivers_path


def delete(device: str | None) -> NoReturn:
    """Remove the driver associated to a device,
    without causing error if the driver does not exists,
    and exit.

    Args:
        device (str | None): path to the infrared camera. None to execute all driver.
    """
    try:
        for driver in get_drivers_path(device):
            os.remove(driver)
    except FileNotFoundError:
        pass  # no driver for this device, but there is no need to send error message

    logging.info("The drivers have been deleted.")
    exit(ExitCode.SUCCESS)

File: command/delete.cpp
#include "commands.hpp"

#include <filesystem>
#include <string>
#include <vector>
using namespace std;

#include "utils/logger.hpp"
#include "configuration.hpp"

/**
 * @brief Deletes the config associated to a device,
 * without causing error if the config does not exists.
 *
 * @param device path to the camera,
 * empty to delete all configs
 *
 * @return exit code
 */
ExitCode delete_config(const char *device)
{
    Logger::debug("Executing delete command.");

    if (string(device).empty())
    {
        auto devices = Configuration::ConfiguredDevices();
        for (const auto &device: devices)
            Configuration::Delete(device);
    }
    else
        Configuration::Delete(device);

    Logger::info("The configurations have been deleted.");
    return ExitCode::SUCCESS;
}
